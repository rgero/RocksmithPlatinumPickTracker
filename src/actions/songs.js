
export const addSong = (song) => ({
    type: 'ADD_SONG',
    song
})

export const removeSong = ( id ) => ({
    type: 'REMOVE_SONG',
    id
})

export const editSong = ( id, update = {}) => ({
    type: "EDIT_SONG",
    id,
    update
})

// SET_SONG
export const setSongs = (songs) => ({
    type: 'SET_SONGS',
    songs
});