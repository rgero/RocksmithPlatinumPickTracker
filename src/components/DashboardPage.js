import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import PickListFilter from './PickListFilters';
import PickList from './PickList';




const DashboardPage = () => (
    (
        <div>
            <div className='content-container'>
                <div className='content-navbar'>
                    <Link className='button' to='/import'>Import Data</Link>
                    <Link className='button' to='/create'>Log New Pick</Link>
                </div>
                <PickListFilter />
                <PickList />
            </div>
        </div>
    )
);

export default DashboardPage;