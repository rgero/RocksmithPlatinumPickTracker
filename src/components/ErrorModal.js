import React from 'react';
import Modal from 'react-modal';

const ErrorModal = (props) =>
    (
        <Modal
            isOpen={!!props.errorMessage}
            contentLabel='Error with Form' // This is for people who have accessiblity options enabled.
            onRequestClose={props.clearError} // This is fired if you click in the background or the escape key.
            closeTimeoutMS={200}
            className='modal'
        >
            {props.errorMessage && 
                <div>
                    <h2>Error</h2>
                    <div>{props.errorMessage}</div>
                    <div><button className='button' onClick={props.clearError}>Close</button></div>
                </div>
            }
        </Modal>
    )

export default ErrorModal