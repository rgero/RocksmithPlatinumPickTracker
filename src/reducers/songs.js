// This is the songs reducer
const songsReducerDefaultState = [];
const songsReducer = (state = songsReducerDefaultState, action) => {
    switch(action.type){
        case 'ADD_SONG':           
            return state.concat(action.song);
        case 'REMOVE_SONG':
            return state.filter(function(testOption){
                return action.id !== testOption.id;
            })
        case 'EDIT_SONG':
            return state.map( (song) => {
                if (song.id === action.id) {
                    return {
                        ...song,
                        ...action.update
                    }
                } else {
                    return song;
                }
            })
        case 'SET_SONGS':
            return action.songs;
        default:
            return state;
    }
};

export default songsReducer;