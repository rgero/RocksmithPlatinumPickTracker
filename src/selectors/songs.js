// How we parse the data.
import moment from 'moment';

// Get visible songs
const getVisibleSongs = (songs, { text, startDate, endDate }) => {
    return songs.filter((song)=> {
        const createdMoment = moment(song.date);
        const startDateMatch = startDate ? startDate.isSameOrBefore(createdMoment, 'day') : true;
        const endDateMatch = endDate ? endDate.isSameOrAfter(createdMoment, 'day') : true;
        const textMatch = typeof text !== "string" || song.artist.toLowerCase().includes(text.toLowerCase()) || song.songName.toLowerCase().includes(text.toLowerCase())

        return startDateMatch && endDateMatch && textMatch;
    })
}

export {getVisibleSongs}