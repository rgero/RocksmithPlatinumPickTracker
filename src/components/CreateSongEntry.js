import React from 'react';
import {connect} from 'react-redux';
import SongForm from './SongForm';
import { addSong } from '../actions/songs';

export class CreateSongEntry extends React.Component {
    constructor(props){
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
    }

    onSubmit(song){
        this.props.onSubmit(song);
        this.props.history.push('/')
    }

    render(){
        return(
            <div>
                <div className="page-header">
                    <div className="content-container">
                        <h1 className="page-header__title">Add Song</h1>
                    </div>
                </div>
                <div className="content-container">
                    <SongForm onSubmit={this.onSubmit} />
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (song) => dispatch(addSong(song))
    }
}

export default connect(undefined, mapDispatchToProps)(CreateSongEntry); // Check out the react-redux documentation to understand the connect statement here.