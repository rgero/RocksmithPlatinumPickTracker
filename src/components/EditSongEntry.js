import React from 'react';
import {connect} from 'react-redux';
import SongForm from './SongForm';
import {startEditSong, startRemoveSong} from '../actions/songs';

export class EditSongPage extends React.Component {
    constructor(props){
        super(props);
        this.onSubmit = this.onSubmit.bind(this);
        this.removeItem = this.removeItem.bind(this);
    }

    onSubmit(song){
        this.props.onSubmit(this.props.song.id, song)
        this.props.history.push('/')
    }

    removeItem(){
        this.props.removeItem(this.props.song.id);
        this.props.history.push('/')
    }

    render(){
        return(
            <div>
                <div className='page-header'>
                    <div className='content-container'>
                        <h1 className='page-header__title'>Editing '{this.props.song.artist} - {this.props.song.songName}'</h1>
                    </div>
                </div>
                <div className='content-container'>
                    <SongForm
                        songInfo = {this.props.song}
                        onSubmit={this.onSubmit}
                        
                    />
                    <button className='button button--secondary' onClick={this.removeItem}>Remove</button>
                </div>
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onSubmit: (id, song) => { 
            dispatch(startEditSong(id, song))
        },
        removeItem: (id) => dispatch(startRemoveSong(id))
    }
}

const mapStateToProps = (state, props) => {
    return {
        song: state.songs.find((song) => song.id === props.match.params.id)
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(EditSongPage);