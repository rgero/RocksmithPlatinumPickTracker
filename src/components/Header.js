import React from 'react';
import { Link } from 'react-router-dom';
import {connect} from 'react-redux';
import {startLogout} from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header className="header">
        <div className="header__content">
            <Link className="header__title" to="/practices" exact="true"><h1>Rocksmith Pick Tracker</h1></Link>
            <button className="button button--link" onClick={startLogout}>Log out</button>
        </div>
    </header>
);

const mapDispatchToProps = (dispatch) => ({

    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header);