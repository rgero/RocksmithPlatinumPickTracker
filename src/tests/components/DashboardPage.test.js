import React from 'react';
import { shallow } from 'enzyme';
import {DashboardPage} from '../../components/DashboardPage';

import songs from '../fixtures/songs';

test("Rendering No Songs", ()=>{
    const wrapper = shallow(<DashboardPage songs={[]}/>);
    expect(wrapper).toMatchSnapshot();
})

test("Rendering Songs", ()=>{
    const wrapper = shallow(<DashboardPage songs={songs}/>);
    expect(wrapper).toMatchSnapshot();
})