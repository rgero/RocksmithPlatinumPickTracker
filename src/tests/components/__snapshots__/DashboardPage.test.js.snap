// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering No Songs 1`] = `
<div>
  <div
    className="content-container"
  >
    <div
      className="content-navbar"
    >
      <Link
        className="button"
        to="/import"
      >
        Import Data
      </Link>
      <Link
        className="button"
        to="/create"
      >
        Log New Pick
      </Link>
    </div>
    <ReactTable
      AggregatedComponent={[Function]}
      ExpanderComponent={[Function]}
      FilterComponent={[Function]}
      LoadingComponent={[Function]}
      NoDataComponent={[Function]}
      PadRowComponent={[Function]}
      PaginationComponent={[Function]}
      PivotValueComponent={[Function]}
      ResizerComponent={[Function]}
      TableComponent={[Function]}
      TbodyComponent={[Function]}
      TdComponent={[Function]}
      TfootComponent={[Function]}
      ThComponent={[Function]}
      TheadComponent={[Function]}
      TrComponent={[Function]}
      TrGroupComponent={[Function]}
      aggregatedKey="_aggregated"
      className="-striped -highlight"
      collapseOnDataChange={true}
      collapseOnPageChange={true}
      collapseOnSortingChange={true}
      column={
        Object {
          "Aggregated": undefined,
          "Cell": undefined,
          "Expander": undefined,
          "Filter": undefined,
          "Footer": undefined,
          "Header": undefined,
          "Pivot": undefined,
          "PivotValue": undefined,
          "Placeholder": undefined,
          "aggregate": undefined,
          "className": "",
          "filterAll": false,
          "filterMethod": undefined,
          "filterable": undefined,
          "footerClassName": "",
          "footerStyle": Object {},
          "getFooterProps": [Function],
          "getHeaderProps": [Function],
          "getProps": [Function],
          "headerClassName": "wordwrap",
          "headerStyle": Object {},
          "minResizeWidth": 11,
          "minWidth": 100,
          "resizable": undefined,
          "show": true,
          "sortMethod": undefined,
          "sortable": undefined,
          "style": Object {},
        }
      }
      columns={
        Array [
          Object {
            "Header": "Your Rocksmith Picks",
            "columns": Array [
              Object {
                "Header": "Artist Name",
                "accessor": "artist",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Song Name",
                "accessor": "songName",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Date",
                "accessor": [Function],
                "id": "date",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Path",
                "accessor": "path",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Difficulty",
                "accessor": "difficulty",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Pick Level",
                "accessor": "level",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Accuracy",
                "accessor": "accuracy",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
            ],
          },
        ]
      }
      data={Array []}
      defaultExpanded={Object {}}
      defaultFilterMethod={[Function]}
      defaultFiltered={Array []}
      defaultPage={0}
      defaultPageSize={10}
      defaultResized={Array []}
      defaultSortDesc={false}
      defaultSortMethod={[Function]}
      defaultSorted={Array []}
      expanderDefaults={
        Object {
          "filterable": false,
          "resizable": false,
          "sortable": false,
          "width": 35,
        }
      }
      filterable={false}
      freezeWhenExpanded={false}
      getLoadingProps={[Function]}
      getNoDataProps={[Function]}
      getPaginationProps={[Function]}
      getProps={[Function]}
      getResizerProps={[Function]}
      getTableProps={[Function]}
      getTbodyProps={[Function]}
      getTdProps={[Function]}
      getTfootProps={[Function]}
      getTfootTdProps={[Function]}
      getTfootTrProps={[Function]}
      getTheadFilterProps={[Function]}
      getTheadFilterThProps={[Function]}
      getTheadFilterTrProps={[Function]}
      getTheadGroupProps={[Function]}
      getTheadGroupThProps={[Function]}
      getTheadGroupTrProps={[Function]}
      getTheadProps={[Function]}
      getTheadThProps={[Function]}
      getTheadTrProps={[Function]}
      getTrGroupProps={[Function]}
      getTrProps={[Function]}
      groupedByPivotKey="_groupedByPivot"
      indexKey="_index"
      loading={false}
      loadingText="Loading..."
      multiSort={true}
      nestingLevelKey="_nestingLevel"
      nextText="Next"
      noDataText="No rows found"
      ofText="of"
      onFetchData={[Function]}
      originalKey="_original"
      pageJumpText="jump to page"
      pageSizeOptions={
        Array [
          5,
          10,
          20,
          25,
          50,
          100,
        ]
      }
      pageText="Page"
      pivotDefaults={Object {}}
      pivotIDKey="_pivotID"
      pivotValKey="_pivotVal"
      previousText="Previous"
      resizable={true}
      resolveData={[Function]}
      rowsSelectorText="rows per page"
      rowsText="rows"
      showPageJump={true}
      showPageSizeOptions={true}
      showPagination={true}
      showPaginationBottom={true}
      showPaginationTop={false}
      sortable={true}
      style={Object {}}
      subRowsKey="_subRows"
    />
  </div>
</div>
`;

exports[`Rendering Songs 1`] = `
<div>
  <div
    className="content-container"
  >
    <div
      className="content-navbar"
    >
      <Link
        className="button"
        to="/import"
      >
        Import Data
      </Link>
      <Link
        className="button"
        to="/create"
      >
        Log New Pick
      </Link>
    </div>
    <ReactTable
      AggregatedComponent={[Function]}
      ExpanderComponent={[Function]}
      FilterComponent={[Function]}
      LoadingComponent={[Function]}
      NoDataComponent={[Function]}
      PadRowComponent={[Function]}
      PaginationComponent={[Function]}
      PivotValueComponent={[Function]}
      ResizerComponent={[Function]}
      TableComponent={[Function]}
      TbodyComponent={[Function]}
      TdComponent={[Function]}
      TfootComponent={[Function]}
      ThComponent={[Function]}
      TheadComponent={[Function]}
      TrComponent={[Function]}
      TrGroupComponent={[Function]}
      aggregatedKey="_aggregated"
      className="-striped -highlight"
      collapseOnDataChange={true}
      collapseOnPageChange={true}
      collapseOnSortingChange={true}
      column={
        Object {
          "Aggregated": undefined,
          "Cell": undefined,
          "Expander": undefined,
          "Filter": undefined,
          "Footer": undefined,
          "Header": undefined,
          "Pivot": undefined,
          "PivotValue": undefined,
          "Placeholder": undefined,
          "aggregate": undefined,
          "className": "",
          "filterAll": false,
          "filterMethod": undefined,
          "filterable": undefined,
          "footerClassName": "",
          "footerStyle": Object {},
          "getFooterProps": [Function],
          "getHeaderProps": [Function],
          "getProps": [Function],
          "headerClassName": "wordwrap",
          "headerStyle": Object {},
          "minResizeWidth": 11,
          "minWidth": 100,
          "resizable": undefined,
          "show": true,
          "sortMethod": undefined,
          "sortable": undefined,
          "style": Object {},
        }
      }
      columns={
        Array [
          Object {
            "Header": "Your Rocksmith Picks",
            "columns": Array [
              Object {
                "Header": "Artist Name",
                "accessor": "artist",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Song Name",
                "accessor": "songName",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Date",
                "accessor": [Function],
                "id": "date",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Path",
                "accessor": "path",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Difficulty",
                "accessor": "difficulty",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Pick Level",
                "accessor": "level",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
              Object {
                "Header": "Accuracy",
                "accessor": "accuracy",
                "style": Object {
                  "whiteSpace": "unset",
                },
              },
            ],
          },
        ]
      }
      data={
        Array [
          Object {
            "accuracy": "100.00",
            "artist": "Rise Against",
            "date": "2020-01-06",
            "difficulty": "Master",
            "id": "1",
            "level": "Bronze",
            "notes": "",
            "path": "Lead",
            "songName": "Savior",
          },
          Object {
            "accuracy": "100.00",
            "artist": "Rise Against",
            "date": "2020-01-06",
            "difficulty": "Master",
            "id": "2",
            "level": "Platinum",
            "notes": "",
            "path": "Bass",
            "songName": "Savior",
          },
          Object {
            "accuracy": "100.00",
            "artist": "Jake's Super Band",
            "date": "2020-01-06",
            "difficulty": "Easy. To strong",
            "id": "3",
            "level": "Silver",
            "notes": "",
            "path": "Lead",
            "songName": "RAPTOR!!",
          },
          Object {
            "accuracy": "100.00",
            "artist": "Bob Marley",
            "date": "2020-01-06",
            "difficulty": "Master",
            "id": "4",
            "level": "Gold",
            "notes": "",
            "path": "Rhythm",
            "songName": "Some Bob Marley Song",
          },
        ]
      }
      defaultExpanded={Object {}}
      defaultFilterMethod={[Function]}
      defaultFiltered={Array []}
      defaultPage={0}
      defaultPageSize={10}
      defaultResized={Array []}
      defaultSortDesc={false}
      defaultSortMethod={[Function]}
      defaultSorted={Array []}
      expanderDefaults={
        Object {
          "filterable": false,
          "resizable": false,
          "sortable": false,
          "width": 35,
        }
      }
      filterable={false}
      freezeWhenExpanded={false}
      getLoadingProps={[Function]}
      getNoDataProps={[Function]}
      getPaginationProps={[Function]}
      getProps={[Function]}
      getResizerProps={[Function]}
      getTableProps={[Function]}
      getTbodyProps={[Function]}
      getTdProps={[Function]}
      getTfootProps={[Function]}
      getTfootTdProps={[Function]}
      getTfootTrProps={[Function]}
      getTheadFilterProps={[Function]}
      getTheadFilterThProps={[Function]}
      getTheadFilterTrProps={[Function]}
      getTheadGroupProps={[Function]}
      getTheadGroupThProps={[Function]}
      getTheadGroupTrProps={[Function]}
      getTheadProps={[Function]}
      getTheadThProps={[Function]}
      getTheadTrProps={[Function]}
      getTrGroupProps={[Function]}
      getTrProps={[Function]}
      groupedByPivotKey="_groupedByPivot"
      indexKey="_index"
      loading={false}
      loadingText="Loading..."
      multiSort={true}
      nestingLevelKey="_nestingLevel"
      nextText="Next"
      noDataText="No rows found"
      ofText="of"
      onFetchData={[Function]}
      originalKey="_original"
      pageJumpText="jump to page"
      pageSizeOptions={
        Array [
          5,
          10,
          20,
          25,
          50,
          100,
        ]
      }
      pageText="Page"
      pivotDefaults={Object {}}
      pivotIDKey="_pivotID"
      pivotValKey="_pivotVal"
      previousText="Previous"
      resizable={true}
      resolveData={[Function]}
      rowsSelectorText="rows per page"
      rowsText="rows"
      showPageJump={true}
      showPageSizeOptions={true}
      showPagination={true}
      showPaginationBottom={true}
      showPaginationTop={false}
      sortable={true}
      style={Object {}}
      subRowsKey="_subRows"
    />
  </div>
</div>
`;
